lib/grammars.coffee

```diff
@@ -134,6 +134,22 @@ module.exports =
       command: "julia"
       args: (context) -> [context.filepath]
 
+  Kotlin:
+    "Selection Based":
+      command: "bash"
+      args: (context) ->
+        code = context.getCode(true)
+        tmpFile = GrammarUtils.createTempFileWithCode(code, ".kt")
+        jarName = context.filename.replace(/\.kt$/g, "") + ".jar"
+        args = ['-c', "kotlinc " + tmpFile + " -include-runtime -d " + jarName + " && java -jar " + jarName + " && rm " + jarName]
+        return args
+    "File Based":
+      command: "bash"
+      args: (context) ->
+        jarName = context.filename.replace(/\.kt$/g, "") + ".jar"
+        args = ['-c', "kotlinc " + context.filepath + " -include-runtime -d " + jarName + " && java -jar " + jarName + " && rm " + jarName]
```

Comment:
For the second argument instead of passing in a huge concatenation of many strings how about using CoffeeScript's built in interpolation?

```coffeescript
"kotlinc #{context.filepath} -include-runtime -d #{jarName}..."
```